PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/webapponlinux/WebAppOnLinuxDeployState.java

```diff
@@ -97,22 +115,40 @@ public ExecutionResult execute(Executor executor, @NotNull ProgramRunner program
                             throw new DockerException(message.toString());
                         }
                     };
-                    PrivateRegistryImageSetting acrInfo = webAppOnLinuxDeployModel.getPrivateRegistryImageSetting();
+                    PrivateRegistryImageSetting acrInfo = deployModel.getPrivateRegistryImageSetting();
                     DockerUtil.pushImage(docker, acrInfo.getServerUrl(), acrInfo.getUsername(), acrInfo.getPassword(),
                             latestImageName, acrInfo.getImageNameWithTag(), progressHandler);
 
-                    // update WebApp
-                    println("Update WebApp ...  ");
-                    WebApp app = AzureWebAppMvpModel.getInstance().updateWebAppOnLinux(webAppOnLinuxDeployModel
-                                    .getSubscriptionId(),
-                            webAppOnLinuxDeployModel.getWebAppId(), acrInfo);
-                    if (app != null && app.name() != null) {
-                        println(String.format("URL:  http://%s.azurewebsites.net/%s", app.name(), project.getName()));
+                    // deploy
+                    if (deployModel.isCreatingNewWebAppOnLinux()) {
+                        // create new WebApp
+                        println("Create new WebApp ...  ");
+                        WebApp app = AzureWebAppMvpModel.getInstance()
+                                .createWebAppOnLinux(deployModel.getSubscriptionId(), deployModel,
+                                        deployModel.getPrivateRegistryImageSetting());
+
+                        if (app != null && app.name() != null) {
+                            println(String.format("URL:  http://%s.azurewebsites.net/%s", app.name(),
+                                    project.getName()));
+                            updateConfigurationDataModel(app);
+                        }
+                    } else {
+                        // update WebApp
+                        println("Update WebApp ...  ");
+                        WebApp app = AzureWebAppMvpModel.getInstance()
+                                .updateWebAppOnLinux(deployModel.getSubscriptionId(), deployModel.getWebAppId(),
+                                        acrInfo);
+                        if (app != null && app.name() != null) {
+                            println(String.format("URL:  http://%s.azurewebsites.net/%s", app.name(),
+                                    project.getName()));
+                        }
                     }
                     return null;
                 }
         ).subscribeOn(Schedulers.io()).subscribe(
```

Comment:
We can directly get the provider from SchedulerProviderFactory: https://github.com/Microsoft/azure-tools-for-java/blob/develop/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebDeployUtil.java#L199